src/net.cpp

```diff
@@ -1781,7 +1813,8 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         //  * Only make a feeler connection once every few minutes.
         //
         bool fFeeler = false;
-        if (nOutbound >= nMaxOutbound) {
+
+        if (nOutbound >= nMaxOutbound && !GetTryNewOutboundPeer()) {
```

Comment:
Yes, I think if `GetTryNewOutboundPeer` is set, then we keep trying outbound peers up to the semaphore count, at which point I believe we block.  Is that problematic for some reason?  I'm not super familiar with the design here but my understanding from reading the code was that this was how this logic was originally written (ie before feelers were introduced, with the `continue` in the `if` block to avoid consuming the last semaphore value via a non-feeler).